lib/checkr.rb

```diff
@@ -61,6 +61,9 @@ def self.api_url(path='')
   end
 
   def self.request(method, path, params={}, headers={})
+    api_key = params[:api_key] ? params[:api_key] : self.api_key
+    params.delete(:api_key)
```

Comment:
Can we add a new method argument instead of overloading params?

Reply:
eg:

```ruby
def self.request(method, path, params={}, headers={}, api_key = nil)
```

Reply:
@jperichon I tried that approach. But all of the non keyword arguments go through pretty tangled chain of methods for composing arguments and it gets pretty messy, additionally this would require changing every single file and explicitly whitelisting that argument as acceptable. This seemed to be an easier solution that would not interfere with the way this gem used to work. What are the considerations against using the current approach?

Reply:
@jperichon Adding a method argument is messy, because none of the methods are actually defined. (they all get created through request method followed by api lambda method that are very abstract level)

Reply:
@jperichon I can try opening another pr with this approach though for comparison.

Reply:
as discussed, ok to move to headers